crates/flux-fhir-analysis/src/conv/mod.rs

```diff
@@ -1961,9 +1961,24 @@ fn conv_sort_path(
     next_infer_sort: &mut impl FnMut() -> rty::Sort,
 ) -> QueryResult<rty::Sort> {
     let ctor = match path.res {
-        fhir::SortRes::PrimSort(fhir::PrimSort::Int) => return Ok(rty::Sort::Int),
-        fhir::SortRes::PrimSort(fhir::PrimSort::Bool) => return Ok(rty::Sort::Bool),
-        fhir::SortRes::PrimSort(fhir::PrimSort::Real) => return Ok(rty::Sort::Real),
+        fhir::SortRes::PrimSort(fhir::PrimSort::Int) => {
+            if !path.args.is_empty() {
+                return Err(emit_prim_sort_generics_error(genv, path, "int", 0))?;
+            }
+            return Ok(rty::Sort::Int);
+        }
+        fhir::SortRes::PrimSort(fhir::PrimSort::Bool) => {
+            if !path.args.is_empty() {
+                return Err(emit_prim_sort_generics_error(genv, path, "bool", 0))?;
+            }
+            return Ok(rty::Sort::Bool);
+        }
+        fhir::SortRes::PrimSort(fhir::PrimSort::Real) => {
+            if !path.args.is_empty() {
+                return Err(emit_prim_sort_generics_error(genv, path, "real", 0))?;
+            }
+            return Ok(rty::Sort::Real);
+        }
         fhir::SortRes::SortParam(n) => return Ok(rty::Sort::Var(rty::ParamSort::from(n))),
         fhir::SortRes::TyParam(def_id) => {
             return Ok(rty::Sort::Param(def_id_to_param_ty(genv, def_id)))
```

Comment:
We need to check here sort parameters do not have arguments. An example where we should report an error:

```rust
#[flux::refined_by(f: T<int>)]
struct S<T> { 
    #[flux::field(T[f])]
    f: T 
}
```

Reply:
This is akin to

```rust
fn foo<T>(x: T<i32>) {}
```